lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -138,11 +158,32 @@ class TwilioConnectionTransport extends StateMachine {
       },
       _userAgent: {
         value: options.userAgent
+      },
+      _wsServer: {
+        value: wsServer
       }
     });
     setupEventListeners(this);
   }
 
+  get _isReconnecting() {
```

Comment:
I think this is misnamed. AFAICT, the underlying TwilioConnection can have been successfully replaced, all Sync/Synced traffic can have completed, and yet `_isReconnecting` will return `true`. To me, this feels more like a mixture of `_hasReconnected` and `_canReconnect`.

Reply:
Looking at how it's used below, actually, could this be as simple as the following?

```js
get _hasReconnected() {
  return this._reconnectAttemptsLeft < this._options.maxReconnectAttempts;
}
```

Reply:
Will `_resetReconnectAttemptsLeft` create issues here, though?

Reply:
@markandrus `_isReconnecting` will only be true if `transport.state` is "connected" and `0 <= transport._reconnectAttemptsLeft < maxReconnectAttempts`. After "synced" message is received, we reset `transport._reconnectAttempts` to `maxReconnectAttempts`. So now, `isReconnecting` will be false.

Reply:
@markandrus Hmm... if we transition to "syncing" right after we get disconnected, then we might not need the `._isReconnecting` property, and the Room#reconnect event will be fired sooner than it is now. Good point. 👍